File: Offer/src/Array/Offer57.java
package Array;

public class Offer57 {
    public int[] twoSum(int[] nums, int target) {
        int left = 0, right = nums.length;
        while (left < right){
            if(nums[left]+nums[right] == target){
                return new int[]{left,right};
            }else if (nums[left]+nums[right] < target){
                left++;
            }else{
                right--;
            }
        }
        return new int[]{};
    }
}

File: Offer/golang/Array/Offer57.go
package array

func twoSum(nums []int, target int) []int {
	left, right := 0, len(nums)-1
	for left < right {
		if nums[left]+nums[right] == target {
			return []int{nums[left], nums[right]}
		} else if nums[left]+nums[right] < target {
			left++
		} else {
			right--
		}
	}
	return []int{}
}
